riscv/processor.h

```diff
@@ -205,20 +205,33 @@ class processor_t : public abstract_device_t
   bool any_custom_extensions() const {
     return !custom_extensions.empty();
   }
+  bool extension_changable(unsigned char ext) const {
+    return (ext == EXT_ZCA &&
+            !(isa->extension_enabled(EXT_ZCB) ||
+              isa->extension_enabled(EXT_ZCMP) ||
+              isa->extension_enabled(EXT_ZCMT))) ||
+           ext == EXT_ZCF || ext == EXT_ZCD;
+  }
   bool extension_enabled(unsigned char ext) const {
     if (ext >= 'A' && ext <= 'Z')
       return state.misa->extension_enabled(ext);
     else
-      return isa->extension_enabled(ext);
+      return isa->extension_enabled(ext) && !extension_disable_table[ext];
```

Comment:
I would prefer that we check only the data structure `extension_disable_table` here, rather than checking two things.  This is better for performance and IMO is easier to reason about.

The implication would be that `extension_enable_table[ext]` always be false when `isa->extension_enabled(ext)` is false.